### PR TITLE
Added files for face detector sample

### PR DESCRIPTION
**Pull request by arrybn**
**Merge with 3rdparty**: https://github.com/opencv/opencv_3rdparty/pull/21

### This pullrequest changes
Added model and sample for a face detector. The detector is based on SSD framework with reduced ResNet-10 backbone.

### CODE REVIEW

File: samples/dnn/resnet_ssd_face_python.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import argparse
+import os
+import sys
+sys.path.append('/home/arrybn/build/opencv/lib')
+import cv2 as cv
+from cv2 import dnn
```
Comment (alalek) [inline samples/dnn/resnet_ssd_face_python.py]: Please avoid local absolute paths.

Consider to add this line into your `.bash_profile` (or `.bashrc`) files:
```
export PYTHONPATH=/home/arrybn/build/opencv/lib
```

and restart your terminal (open new one) or call `source ~/.bash_profile`

Comment (arrybn) [reply samples/dnn/resnet_ssd_face_python.py]: removed this from the script

**Comment by alalek** [comment]
Perhaps we need python-based model downloader too, something like this: https://github.com/alalek/opencv/commit/pr9516

File: samples/dnn/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,28 @@ ocv_check_dependencies(${OPENCV_DNN_SAMPLES_REQUIRED_DEPS})
...
 
+  # Model branch name: dnn_samples_face_detector_20170830
+  set(DNN_FACE_DETECTOR_MODEL_COMMIT "b2bfc75f6aea5b1f834ff0f0b865a7c18ff1459f")
+  set(DNN_FACE_DETECTOR_MODEL_HASH "afbb6037fd180e8d2acb3b58ca737b9e")
+
+  set(DNN_FACE_DETECTOR_MODEL_NAME "res10_300x300_ssd_iter_140000.caffemodel")
+
+  set(DNN_FACE_DETECTOR_MODEL_DOWNLOAD_DIR "${CMAKE_CURRENT_LIST_DIR}/face_detector")
+
+  ocv_download(FILENAME ${DNN_FACE_DETECTOR_MODEL_NAME}
```
Comment (alalek) [inline samples/dnn/CMakeLists.txt]:
```
if(COMMAND ocv_download)
  ocv_download(...)
endif()
```

Comment (arrybn) [reply samples/dnn/CMakeLists.txt]: Fixed

**Comment by alalek** [comment]
:+1:

**Comment by thmo** [comment]
@arrybn as a follow-up for #19290, could you please clarify the license this contribution is under?

**Pull request merged**